aesara/link/numba/dispatch.py

```diff
@@ -1804,3 +1814,24 @@ def batched_dot(x, y):
 # NOTE: The remaining `aesara.tensor.blas` `Op`s appear unnecessary, because
 # they're only used to optimize basic `Dot` nodes, and those GEMV and GEMM
 # optimizations are apparently already performed by Numba
+
+
+@numba_funcify.register(RandomVariable)
+def numba_funcify_RandomVariable(op, node, **kwargs):
+    name = op.name
+    random_func = getattr(np.random, name)
+
+    tuple_type = numba.types.UniTuple(numba.types.intp, node.inputs[1].data.size)
+
+    @numba.njit
+    def random_variable(rng, size, dtype, *args):
+        with numba.objmode(size=tuple_type):
+            size = tuple(size)
```

Comment:
We should be able to avoid this costly jump to object mode by using `numba.np.unsafe.ndarray.to_fixed_tuple` or `create_tuple_creator` and `tuple_setitem`.  See [`numba_funcify_Reshape`](https://github.com/pymc-devs/aesara/blob/abf6026c9c2bad19216275e127e3043f30d93360/aesara/link/numba/dispatch.py#L965) for an example of the latter.

Reply:
I just tried changing `numba_funcify_Reshape` so that it uses `to_fixed_tuple`, and that approach works well&mdash;while also being a lot more succinct.

Reply:
[Here's a PR](https://github.com/pymc-devs/aesara/pull/413) with those changes.